crates/stackable-webhook/src/servers/conversion.rs

```diff
@@ -19,141 +66,246 @@ where
+    ///             .expect("failed to merge S3Connection CRD"),
+    ///         S3Connection::try_convert as fn(_) -> _,
+    ///     ),
+    /// ];
+    ///
+    /// const OPERATOR_NAME: &str = "PRODUCT_OPERATOR";
+    /// let client = Client::try_default().await.expect("failed to create Kubernetes client");
+    /// // Normally you would get this from the CLI arguments in
+    /// // `ProductOperatorRun::operator_environment`
+    /// let operator_environment = OperatorEnvironmentOpts {
+    ///     operator_namespace: "stackable-operator".to_string(),
+    ///     operator_service_name: "product-operator".to_string(),
+    /// };
     ///
     /// // Construct the conversion webhook server
-    /// let server = ConversionWebhookServer::new(handler, Options::default());
+    /// let conversion_webhook = ConversionWebhookServer::new(
+    ///     crds_and_handlers,
+    ///     stackable_webhook::Options::default(),
+    ///     client,
+    ///     OPERATOR_NAME,
+    ///     operator_environment,
+    /// )
+    /// .await
+    /// .expect("failed to create ConversionWebhookServer");
     ///
-    /// // Define the handler function
-    /// fn handler(req: ConversionReview) -> ConversionReview {
-    ///    // In here we can do the CRD conversion
-    ///    req
-    /// }
+    /// conversion_webhook.run().await.expect("failed to run ConversionWebhookServer");
+    /// # }
     /// ```
-    #[instrument(name = "create_conversion_webhhok_server", skip(handler))]
-    pub fn new<H>(handler: H, options: Options) -> Self
+    #[instrument(
+        name = "create_conversion_webhook_server",
+        skip(crds_and_handlers, client)
+    )]
+    pub async fn new<H>(
+        crds_and_handlers: impl IntoIterator<Item = (CustomResourceDefinition, H)>,
+        options: Options,
+        client: Client,
+        field_manager: impl Into<String> + Debug,
+        operator_environment: OperatorEnvironmentOpts,
+    ) -> Result<Self, ConversionWebhookError>
     where
         H: WebhookHandler<ConversionReview, ConversionReview> + Clone + Send + Sync + 'static,
     {
         tracing::debug!("create new conversion webhook server");
+        let field_manager: String = field_manager.into();
 
-        let handler_fn = |Json(review): Json<ConversionReview>| async {
-            let review = handler.call(review);
-            Json(review)
-        };
+        let mut router = Router::new();
+        let mut crds = Vec::new();
+        for (crd, handler) in crds_and_handlers {
+            let crd_name = crd.name_any();
+            let handler_fn = |Json(review): Json<ConversionReview>| async {
+                let review = handler.call(review);
+                Json(review)
+            };
 
-        let router = Router::new().route("/convert", post(handler_fn));
-        Self { router, options }
-    }
+            router = router.route(&format!("/convert/{crd_name}"), post(handler_fn));
```

Comment:
suggestion: Split this.

```suggestion
            let route = format!("/convert/{crd_name}, crd_name = crd.name_any());
            router = router.route(&route), post(handler_fn));
```

Reply:
Hope f1fee5d works for you